packages/tools/client-logger/app-insights-logger/src/FluidAppInsightsLogger.ts

```diff
@@ -0,0 +1,47 @@
+import { ApplicationInsights, Snippet } from "@microsoft/applicationinsights-web";
+
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+
+export interface FluidAppInsightsLoggerConfig {
+	appInsights: ApplicationInsights | Snippet;
+}
+
+export class FluidAppInsightsLogger implements ITelemetryBufferedLogger {
+	protected readonly baseLoggingClient: ApplicationInsights;
+
+	/**
+	 * Creates an instance of FluidAppInsightsLogger using either a provided instance of ApplicationInsights
+	 * logging client or by creating a new instance with a provided ApplicationInsights configuration object.
+	 * @param config - Accepts either an existing preconfigured ApplicationInsights instance to use
+	 * for logging or a config object used to instantiate an instance of ApplicationInsights.
+	 */
+	public constructor(config: FluidAppInsightsLoggerConfig) {
+		if (config.appInsights instanceof ApplicationInsights) {
+			this.baseLoggingClient = config.appInsights;
+		} else {
```

Comment:
I like the idea if the union type. Some thoughts, though:

- Can we use it as the argument to the constructor directly and do away with the extra interface?
- I think we need something more than the `instanceof` check. If the consumer is allowed to give us the instance, they could be using a different version of `@microsoft/applicationinsights-web` than the one we depend on, in which case that check will return false because the types are technically different since they come from different npm packages (or different versions of the same package). We'll probably want something like what TypeScript's [type-guard methods](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) usually do, like checking that the object has a known public property of the `ApplicationInsights` type that we'd expect to be present across all versions we support.

Reply:
Updated this to be real simple, just accepts the app insights client in commit 70aa9fd